src/processTargets/modifiers/ItemStage/ItemStage.ts

```diff
@@ -0,0 +1,169 @@
+
+    const itemInfoWithIntersections = filterItemInfos(target, itemInfos);
+
+    if (itemInfoWithIntersections.length === 0) {
+      throw new NoContainingScopeError(this.modifier.scopeType.type);
+    }
+
+    itemInfoWithIntersections.sort(
+      (a, b) =>
+        rangeLength(target.editor, b.intersection!) -
+        rangeLength(target.editor, a.intersection!)
+    );
+
+    return this.itemInfoToTarget(target, itemInfoWithIntersections[0].itemInfo);
+  }
+
+  private itemInfoToTarget(target: Target, itemInfo: ItemInfo) {
+    const delimiter = getInsertionDelimiter(
+      target.editor,
+      itemInfo.leadingDelimiterRange,
+      itemInfo.trailingDelimiterRange,
+      ", "
+    );
+    return new ScopeTypeTarget({
+      scopeTypeType: <SimpleScopeTypeType>this.modifier.scopeType.type,
+      editor: target.editor,
+      isReversed: target.isReversed,
+      contentRange: itemInfo.contentRange,
+      delimiter,
+      leadingDelimiterRange: itemInfo.leadingDelimiterRange,
+      trailingDelimiterRange: itemInfo.trailingDelimiterRange,
+    });
+  }
+}
+
+/** Filter item infos by content range and domain intersection */
+function filterItemInfos(target: Target, itemInfos: ItemInfo[]) {
+  return itemInfos
+    .map((itemInfo) => ({
+      itemInfo,
+      intersection: itemInfo.domain.intersection(target.contentRange),
+    }))
+    .filter((e) => e.intersection != null);
+}
+
+function getItemInfosForIterationScope(
+  context: ProcessedTargetsContext,
+  target: Target
+) {
+  const { range, boundary } = getIterationScope(context, target);
+  return rangeToItemInfos(target.editor, range, boundary);
+}
+
+function rangeToItemInfos(
+  editor: TextEditor,
+  collectionRange: Range,
+  collectionBoundary?: [Range, Range]
+): ItemInfo[] {
+  const tokens = tokenizeRange(editor, collectionRange, collectionBoundary);
+  const itemInfos: ItemInfo[] = [];
+
+  tokens.forEach((token, i) => {
+    if (token.type === "delimiter" || token.type === "boundary") {
+      return;
+    }
+    const leadingDelimiterRange = (() => {
+      if (tokens[i - 2]?.type === "item") {
+        return new Range(tokens[i - 2].range.end, token.range.start);
+      }
+      if (tokens[i - 1]?.type === "delimiter") {
+        return new Range(tokens[i - 1].range.start, token.range.start);
+      }
+      return undefined;
+    })();
+    const trailingDelimiterRange = (() => {
+      if (tokens[i + 2]?.type === "item") {
+        return new Range(token.range.end, tokens[i + 2].range.start);
+      }
+      if (tokens[i + 1]?.type === "delimiter") {
+        return new Range(token.range.end, tokens[i + 1].range.end);
+      }
+      return undefined;
+    })();
+    // Leading boundary is excluded and leading delimiter is included
+    const leadingMatchStart =
+      tokens[i - 1]?.type === "boundary"
+        ? tokens[i - 1].range.end
+        : tokens[i - 1]?.type === "delimiter"
+        ? tokens[i - 1].range.start
```

Comment:
I believe this should be
```suggestion
        ? tokens[i - 1].range.end
```

The fact that this ends up going left when cursor is directly before `,` happens in a fairly opaque way; both this item and the proceeding one will match the domain, and it's just the fact that the previous item ends up having its match earlier in the list of matches that causes it to get returned

Reply:
No it is correct afaict. The separator is included in the leading range and excluded in the trailing range. I'm actually sorting them on intersection range length to always get the correct one.  
itemInfoWithIntersections.sort(
      (a, b) =>
        rangeLength(target.editor, b.intersection!) -
        rangeLength(target.editor, a.intersection!)
    );